packages/opencensus-exporter-stackdriver/src/stackdriver-cloudtrace-utils.ts

```diff
@@ -0,0 +1,170 @@
+  return attributesBuilder;
+}
+
+/**
+ * Creates StackDriver TimeEvents from OpenCensus Annotation and MessageEvent.
+ * @param annotationTimedEvents coreTypes.Annotation[]
+ * @param messageEventTimedEvents coreTypes.MessageEvent[]
+ * @param droppedAnnotationsCount number
+ * @param droppedMessageEventsCount number
+ * @returns types.TimeEvents
+ */
+export function createTimeEvents(
+    annotationTimedEvents: coreTypes.Annotation[],
+    messageEventTimedEvents: coreTypes.MessageEvent[],
+    droppedAnnotationsCount: number,
+    droppedMessageEventsCount: number): types.TimeEvents {
+  const timeEvents: types.TimeEvent[] = [];
+  if (annotationTimedEvents) {
+    annotationTimedEvents.forEach(annotation => {
+      timeEvents.push({
+        time: new Date(annotation.timestamp).toISOString(),
+        annotation: {
+          description: stringToTruncatableString(annotation.description),
+          attributes: createAttributesBuilder(annotation.attributes, 0)
+        }
+      });
+    });
+  }
+  if (messageEventTimedEvents) {
+    messageEventTimedEvents.forEach(messageEvent => {
+      timeEvents.push({
+        time: new Date(messageEvent.timestamp).toISOString(),
+        messageEvent: {
+          id: messageEvent.id,
+          type: createMessageEventType(messageEvent.type)
+        }
+      });
+    });
+  }
+  return {
+    timeEvent: timeEvents,
+    droppedAnnotationsCount,
+    droppedMessageEventsCount
+  };
+}
+
+export function stringToTruncatableString(value: string):
+    types.TruncatableString {
+  return {value};
+}
+
+export async function getResourceLabels(
+    monitoredResource: Promise<types.MonitoredResource>) {
+  const resource = await monitoredResource;
+  const resourceLabels: Record<string, types.AttributeValue> = {};
+  if (resource.type === 'global') {
+    return resourceLabels;
+  }
+  for (const key of Object.keys(resource.labels)) {
+    const resourceLabel = `g.co/r/${resource.type}/${key}`;
+    resourceLabels[resourceLabel] = createAttributeValue(resource.labels[key]);
+  }
+  return resourceLabels;
+}
+
+function createAttributesBuilder(
+    attributes: coreTypes.Attributes,
+    droppedAttributesCount: number): types.Attributes {
+  const attributeMap: Record<string, types.AttributeValue> = {};
+  for (const key of Object.keys(attributes)) {
+    attributeMap[key] = createAttributeValue(attributes[key]);
+  }
+  return {attributeMap, droppedAttributesCount};
+}
+
+function createLink(link: coreTypes.Link): types.Link {
+  const traceId = link.traceId;
+  const spanId = link.spanId;
+  const type = createLinkType(link.type);
+  const attributes = createAttributesBuilder(link.attributes, 0);
+  return {traceId, spanId, type, attributes};
+}
+
+function createAttributeValue(value: string|number|
+                              boolean): types.AttributeValue {
+  switch (typeof value) {
+    case 'number':
+      return {intValue: String(value)};
```

Comment:
Could you make this be `doubleValue`? The `number` type in JS is a double so we would be losing accuracy for floating point values.

I would think if something is a BigInt and fits in 64 bits it should be made to be an `intValue`, which it looks like [Node 10 supports](https://github.com/nodejs/node/blob/master/doc/changelogs/CHANGELOG_V10.md#2018-10-10-version-10120-current-targos)

Reply:
v2 API doesn't support ```doubleValue``` type for ```AttributeValue``` -> https://github.com/googleapis/google-api-nodejs-client/blob/master/src/apis/cloudtrace/v2.ts#L153-L166

Reply:
Hmm, OK. Then here are a couple options:

* Just always translate `number` to its string representation. This is probably my suggestion. Maybe add a TODO to change this once the API supports it. The OpenCensus trace proto has `doubleValue`.
* Check whether the number fits in a 64-bit int (less than the max 64-bit int and no fractional part) and only then write it as an `intValue` and otherwise use `String`.

I think `intValue` uses `String` type because `number` can't represent full accuracy 64-bit ints.

Reply:
Added TODO here to consider ```doubleValue``` type once available with cloudTrace V2 API. As per our offline discussion (+ your suggestion), will always translate ```number``` to its string representation for ```AttributeValue``` type.